src/client/auth/plain.rs

```diff
@@ -23,12 +27,15 @@ pub(crate) async fn authenticate_stream(
         .as_ref()
         .ok_or_else(|| Error::authentication_error("PLAIN", "no password supplied"))?;
 
-    let sasl_start = SaslStart::new(
+    let mut sasl_start = SaslStart::new(
```

Comment:
It might make sense to add `server_api` as a parameter to `SaslStart` and `SaslContinue` to reduce the repeated code on each call.